src/ts/ConvivaAnalytics.ts

```diff
@@ -488,12 +510,63 @@ export class ConvivaAnalytics {
   release(): void {
     this.unregisterPlayerEvents();
     this.endSession();
-    this.client.releasePlayerStateManager(this.playerStateManager);
     this.client.release();
     this.systemFactory.release();
   }
 }
 
+class PlayerConfigHelper {
+  /**
+   * The config for autoplay and preload have great impact to the VST (Video Startup Time) we track it.
+   * Since there is no way to get default config values from the player they are hardcoded.
+   */
+  static AUTOPLAY_DEFAULT_CONFIG: boolean = false;
+
+  /**
+   * Extract autoplay config form player
+   *
+   * @param player: Player
+   */
+  public static getAutoplayConfig(player: Player): boolean {
+    let playerConfig = player.getConfig();
+
+    if (playerConfig.playback && playerConfig.playback.autoplay !== undefined) {
+      return playerConfig.playback.autoplay;
+    } else {
+      return PlayerConfigHelper.AUTOPLAY_DEFAULT_CONFIG;
+    }
+  }
+
+  /**
+   * Extract preload config from player
+   *
+   * The preload config can be set individual for mobile or desktop.
+   * Default value is true for VOD and false for live streams. If the value is not set for current platform
+   * the default value will be used over the value for the other platform.
+   *
+   * @param player: Player
+   */
+  public static getPreloadConfig(player: Player): boolean {
```

Comment:
The preload can also be set on root level: 
```
playerConfig.adaptation.preload
```

Reply:
Thanks. Didn't know that.

Reply:
That wasn't necessary. The player setup expands a root level setting into the `mobile` and `desktop` subsettings.